frontend/components/service/addServiceForm.js

```diff
@@ -1,190 +1,182 @@
-import { Grid } from '@mui/material';
+import {useEffect, useState} from 'react';
+import {Grid} from '@mui/material';
 import ServiceEmployeeTable from './serviceEmployeeTable';
 import DurationPrice from './durationPrice';
 import AddIcon from '@mui/icons-material/Add';
+import {v4 as uuidv4} from 'uuid';
+import {InputTextField} from "../form/formComponent";
 
 const AddServiceForm = (props) => {
-  const hourToMs = 60000;
-  const MsToHour = 3600000;
-  const { addHandle, employeeList, open, closeDialog } = props;
-  const [employeeCheckList, setEmployeeCheckList] = useState([]);
-  const [name, setName] = useState('');
-  const [barcode, setBarcode] = useState('');
-  const [description, setDescription] = useState('');
-  const [duration, setDuration] = useState(0);
-  const [price, setPrice] = useState(0);
-  const [durationPriceList, setDurationPriceList] = useState([
-    { price: 0, duration: 0.5 },
-  ]);
-  const [reload, setReload] = useState(false);
+    const initValue = {
+        service_code: '',
+        name: '',
+        description: '',
+        treatment_type: '',
+        barcode: '',
+        sms_description: ''
+    };
+    const mshconvertion = 3600000;
+    const {addHandle, employeeList, open, closeDialog} = props;
+    const [employeeCheckList, setEmployeeCheckList] = useState([]);
+    const [durationPriceList, setDurationPriceList] = useState([]);
+    const [reload, setReload] = useState(false);
+    const [serviceValue, setServiceValue] = useState(initValue);
+    const [errorMessage, setErrorMessage] = useState({});
 
-  const processAddService = () => {
-    if (validationInput()) {
-      let data = {
-        service_code: barcode,
-        name: 'service ' + barcode,
-        description: description,
-        treatment_type: 'type 1',
-        duration: 3600000,
-        price: 120,
-        barcode: barcode,
-        sms_description: 'sms description 1',
-      };
-      addHandle(data);
-    }
-  };
-  const handleAddEmployeeCheck = (val, employee) => {
-    if (val) {
-      setEmployeeCheckList([...employeeCheckList, employee]);
-    } else {
-      setEmployeeCheckList(
-        employeeCheckList.filter((emp) => emp.firstname != employee.firstname)
-      );
+    const handleSetServiceValue = (obj) => {
+        const {name, value} = obj.target;
+        setServiceValue({...serviceValue, [name]: value});
+    };
+
+    const validate = () => {
+        let temp = {}
+        temp.name = serviceValue.name ? "" : "This field is required.";
+        temp.durationprice = durationPriceList.length > 0 ? "" : "This field is required.";
+        setErrorMessage(temp);
+        return Object.values(temp).every(x => x == "")
     }
-  };
 
-  useEffect(() => {}, [employeeCheckList, durationPriceList, reload]);
+    const processAddService = () => {
+        if (validate()) {
+            //convert the duration to ms
+            let durationTemp = durationPriceList.map(d => (
```

Comment:
Consider inlining the variables (a refactoring practice) in lines 51 and 52. See https://refactoring.com/catalog/inlineVariable.html

Reply:
ok, updated